### PR TITLE
youtube-music: init at 1.17.0

### PR DESCRIPTION
**Pull request by aacebedo**
###### Description of changes

Youtube-music is a electron based app enabling to access youtube music through a native app

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/youtube-music/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ lib, fetchurl, appimageTools, }:
+
+let
+  pname = "youtube-music";
+  version = "1.17.0";
+  name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/youtube-music/default.nix:6]:
```suggestion
```
no longer required

File: pkgs/applications/audio/youtube-music/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/th-ch/youtube-music/releases/download/v${version}/YouTube-Music-${version}.AppImage";
+    sha256 = "sha256-3QNmCek6UD0WNaqalIksYWLHGfMgetLRLi8mEM1J39I=";
+  };
+
+  appimageContents = appimageTools.extract { inherit name src; };
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/youtube-music/default.nix:13]:
```suggestion
  appimageContents = appimageTools.extract { inherit pname src; };
```

File: pkgs/applications/audio/youtube-music/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+in appimageTools.wrapType2 rec {
+  inherit name src;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/youtube-music/default.nix:15]:
```suggestion
  inherit pname src;
```

File: pkgs/applications/audio/youtube-music/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  extraInstallCommands = ''
+    mv $out/bin/{${name},${pname}}
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/youtube-music/default.nix:18]:
```suggestion
    mv $out/bin/{${pname}-${version},${pname}}
```

File: pkgs/applications/audio/youtube-music/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    ls ${appimageContents}
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/youtube-music/default.nix:19]:
```suggestion
```

File: pkgs/applications/audio/youtube-music/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    substituteInPlace \
+        $out/share/applications/youtube-music.desktop \
+        --replace 'Exec=AppRun' 'Exec=${pname}'
+    cp -r ${appimageContents}/usr/share/icons $out/share
+  '';
+
+  meta = with lib; {
+    description = "Electron wrapper around YouTube Music";
+    homepage = "https://th-ch.github.io/youtube-music/";
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/youtube-music/default.nix:32]: sourceProvenance is missing

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/youtube-music/default.nix (hunk 6 above)
Comment (aacebedo) [reply pkgs/applications/audio/youtube-music/default.nix:31]: done

**Comment by aacebedo** [comment]
thanks @SuperSandro2000 for the comments. I applied them and repushed it.

**Pull request merged**